gcloud/monitoring/metric.py

```diff
@@ -0,0 +1,241 @@
+        ``ValueType.INT64``, ``ValueType.DOUBLE``, ``ValueType.STRING``,
+        ``ValueType.DISTRIBUTION``, or ``ValueType.MONEY``.
+        See :class:`ValueType`.
+
+    :type unit: string
+    :param unit: The unit in which the metric value is reported.
+
+    :type description: string
+    :param description: A detailed description of the metric.
+
+    :type display_name: string
+    :param display_name: A concise name for the metric.
+    """
+
+    def __init__(self, name, type, labels, metric_kind, value_type,
+                 unit, description, display_name):
+        # Allow "type" as a parameter name: pylint: disable=redefined-builtin
+        self.name = name
+        self.type = type
+        self.labels = labels
+        self.metric_kind = metric_kind
+        self.value_type = value_type
+        self.unit = unit
+        self.description = description
+        self.display_name = display_name
+
+    @classmethod
+    def _fetch(cls, client, metric_type):
+        """Look up a metric descriptor by type.
+
+        :type client: :class:`gcloud.monitoring.client.Client`
+        :param client: The client to use.
+
+        :type metric_type: string
+        :param metric_type: The metric type name.
+
+        :rtype: :class:`MetricDescriptor`
+        :returns: The metric descriptor instance.
+
+        :raises: :class:`gcloud.exceptions.NotFound`
+        """
+        path = '/projects/{project}/metricDescriptors/{type}'.format(
+            project=client.project,
+            type=metric_type)
+        info = client.connection.api_request(method='GET', path=path)
+        return cls._from_dict(info)
+
+    @classmethod
+    def _list(cls, client, filter=None):
+        """List all metric descriptors for the project.
+
+        :type client: :class:`gcloud.monitoring.client.Client`
+        :param client: The client to use.
+
+        :type filter: string or None
+        :param filter: An optional filter string describing the metric
+            descriptors to be returned. See the `filter documentation`_.
+
+        :rtype: list of :class:`MetricDescriptor`
+        :returns: A list of metric descriptor instances.
+
+        .. _filter documentation:
+            https://cloud.google.com/monitoring/api/v3/filters
+        """
+        # Allow "filter" as a parameter name: pylint: disable=redefined-builtin
+
+        path = '/projects/{project}/metricDescriptors/'.format(
+            project=client.project)
+
+        descriptors = []
+
+        page_token = None
+        while True:
+            params = {}
+
+            if filter is not None:
+                params['filter'] = filter
+
+            if page_token is not None:
+                params['pageToken'] = page_token
+
+            response = client.connection.api_request(
+                method='GET', path=path, query_params=params)
+            for info in response.get('metricDescriptors', []):
```

Comment:
Use `response.get('metricDescriptors', ())` since a `tuple` uses less memory. (Honestly not even approaching a big deal if the code is left as is, but a general idea that is good to keep in mind.)